packages/block-editor/src/components/block-preview/index.js

```diff
@@ -33,19 +37,57 @@ function BlockPreview( props ) {
 }
 
 export function BlockPreviewContent( { blocks, settings } ) {
+	const previewRef = useRef( null );
+
+	const [ previewScale, setPreviewScale ] = useState( 1 );
+	const [ visibility, setVisibility ] = useState( 'hidden' );
+
+	let timerId;
+
+	// Dynamically calculate the scale factor
+	useLayoutEffect( () => {
+		const { clientId } = blocks;
```

Comment:
`blocks` can be an array. This approach won't work in that situation.

Reply:
yes, I think if the blocks is an array (template), we should just scale to fit the whole width somehow.

Reply:
I've resolved this by looping all the Blocks. We'll need to watch out for perf concerns on this one though.